localization/ekf_localizer/src/ekf_localizer.cpp

```diff
@@ -632,7 +648,17 @@ void EKFLocalizer::initSimple1DFilters(const geometry_msgs::msg::PoseWithCovaria
   roll_filter_.init(rpy.x, roll_dev, pose.header.stamp);
   pitch_filter_.init(rpy.y, pitch_dev, pose.header.stamp);
 }
+double EKFLocalizer::updateZConsideringDelay(
+  const geometry_msgs::msg::TwistStamped & twist, const double delay_time)
+{
+  const double vx = twist.twist.linear.x;
+  const double pitch_rad = pitch_from_ndt_;
+  const double val_sin = std::sin(-pitch_rad);
 
+  const double dz = val_sin * vx * delay_time;
+
+  return dz;
+}
```

Comment:
How about something like `calculateDeltaZFromPitch` ?
The current function name sounds a bit different from what it actually does :+1: